Assets/HoloToolkit/SpatialMapping/Scripts/SpatialMappingManager.cs

```diff
@@ -48,7 +50,35 @@ public partial class SpatialMappingManager : Singleton<SpatialMappingManager>
         /// <summary>
         /// The current source of spatial mapping data.
         /// </summary>
-        public SpatialMappingSource Source { get; private set; }
+        public SpatialMappingSource Source
+        {
+            get { return source; }
+
+            private set
+            {
+                if (source != value)
+                {
+                    UpdateRendering(false);
+
+                    var oldSource = source;
+                    source = value;
+
+                    UpdateRendering(DrawVisualMeshes);
+
+                    var handlers = SourceChanged;
+                    if (handlers != null)
+                    {
+                        handlers(this, PropertyChangedEventArgsEx.Create(() => Source, oldSource, source));
```

Comment:
I believe the HoloLens Unity Performance Recommendations said to avoid lambdas, because they implicitly create an extra GC object if the lambda captures any locals, but in this case, the lambda doesn't capture any locals.  I'll see if I can get that recommendation explained more on the page.